edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/config/TwinConfigSource.cs

```diff
@@ -186,6 +230,42 @@ Task UpdateReportedProperties(long desiredVersion, LastDesiredStatus desiredStat
             }
         }
 
+        private static bool CheckIfTwinPropertiesAreSigned(TwinCollection twinDesiredProperties) => JObject.Parse(twinDesiredProperties.ToString())["integrity"] != null;
+
+        public static bool ExtractHubTwinAndVerify(TwinCollection twinDesiredProperties)
+        {
+            try
+            {
+                // Extract Desired properties
+                JObject desiredProperties = new JObject();
+                JObject twinJobject = JObject.Parse(twinDesiredProperties.ToString());
```

Comment:
Similar comment as Agent - you should able to parse this into the actual object.